pkg/drivers/pgsql/pgsql.go

```diff
@@ -153,9 +154,10 @@ func createDBIfNotExist(dataSourceName string) error {
 		logrus.Tracef("SETUP EXEC : %v", util.Stripped(stmt))
 		_, err = db.Exec(stmt)
 		if err != nil {
-			return err
+			logrus.Warnf("failed to create database %s: %v", dbName, err)
+		} else {
+			logrus.Tracef("created database: %s", dbName)
 		}
-		logrus.Tracef("created database: %s", dbName)
 	}
 	return nil
```

Comment:
if we never return any errors, should this function even have a return value?

Reply:
We return an error on line 133 if we can't parse the `dataSourceName`. Although, with the current code preceding the call site, `createDBIfNotExist` can never be called with an invalid `dataSourceName`, so that error will never happen. But it's good to handle it anyway, in case the calling function changes.